crates/interpreter/src/interpreter/stack.rs

```diff
@@ -215,10 +226,15 @@ impl Stack {
     pub fn push(&mut self, value: U256) -> bool {
         // In debug builds, verify we have sufficient capacity provisioned.
         debug_assert!(self.data.capacity() >= STACK_LIMIT);
-        if self.data.len() == STACK_LIMIT {
+        let len = self.data.len();
+        if len == STACK_LIMIT {
             return false;
         }
-        self.data.push(value);
+        unsafe {
+            let end = self.data.as_mut_ptr().add(len);
+            core::ptr::write(end, value);
+            self.data.set_len(len + 1);
+        }
```

Comment:
This is nice! we didn't optimize this function as it is not used that much, PUSH opcodes use optimized push_slice